src/servers/grpc_server.cc

```diff
@@ -3448,8 +3477,9 @@ ModelStreamInferHandler::StartNewRequest()
   state->trace_manager_ = nullptr;
   state->trace_ = nullptr;
   state->trace_id_ = 0;
+  state->trace_userp_ = nullptr;
   if (trace_manager_ != nullptr) {
-    state->trace_ = trace_manager_->SampleTrace();
+    state->trace_ = trace_manager_->SampleTrace(&state->trace_userp_);
```

Comment:
Correct me if I am wrong, isn't `trace_userp`  passed to `SampleTrace` always nullptr?

Reply:
Note that the pointer to the class member is passed, so it will not be nullptr. This is similar to `TRITONBACKEND_InputProperties` where you will pass an non-nullptr if you want to know the property associated with the object, here is the userp which the trace manager newed.

For HTTP frontend, it will be nullptr because the HTTP frontend doesn't need to release the trace and related data directly. The trace object will only be created when an inference request is received and ownership of the trace object will be passed into server core.

Reply:
Ahh.. Makes sense!